### ISSUE TITLE
To be able to use multiple sprites or models in a particle system

### ISSUE THREAD

**Issue by yaustar** [playcanvas/engine#3400]
(Request from client)

**Comment by mvaligursky** [comment]
Is this what is requested?
http://playcanvas.github.io/#/graphics/particles-random-sprites

**Comment by yaustar** [comment]
Not quite. They were hoping to be able to use the sprite asset type with a non grid based texture atlas. In the engine example, each frame is the same size and have the same number of frames.

They would also need to be able to preview in the Editor too.

**Comment by marcusx2** [comment]
I like the idea. I've had this use case before, where I needed a particle system with different particles, not using the same one. The workaround is to have multiple particle systems, but if I was able to use just one it'd be better.

**Comment by mvaligursky** [comment]
The existing examples cover part of this request:

- [particles-anim-index](https://playcanvas.github.io/#/graphics/particles-anim-index) selects a fixed animation from a regular sprite sheet for each particle system. The example uses four separate systems.
- [particles-random-sprites](https://playcanvas.github.io/#/graphics/particles-random-sprites) mixes different appearances within a single particle system using `randomizeAnimIndex`. It demonstrates both four different six-frame coin animations and seven different static bonus images (`animNumFrames: 1`).

So different-looking particles within one system are already supported when their images are packed into a regular grid in one texture. The current animation settings use equally sized grid cells and a shared frame count for all animations.

Following the clarification above, the remaining possibilities to consider are:

- Using actual Sprite assets and their Texture Atlas frame definitions, including non-grid atlas layouts and differently sized frame regions.
- Supporting animations with different frame counts within the same particle system.
- Providing Editor authoring and preview support for those Sprite asset workflows.
- Selecting between multiple meshes/models within one particle system, as mentioned in the title. Currently, particles share a single mesh.

It may be worth narrowing the issue description to these remaining capabilities, since the basic case of mixing images or equal-length sprite-sheet animations is already implemented.
